docs/proxy/datacenter_proxy/examples.md

```diff
@@ -0,0 +1,700 @@
+    }
+});
+</marked-tab>
+```
+
+
+### [](#single-ip-address-from-a-group) Single IP address from a group
+
+Using [PuppeteerCrawler](https://sdk.apify.com/docs/api/puppeteer-crawler#docsNav), get a new IP address selected from the `SHADER` proxy group for each browser opened during an entire run.
+
+If the IP fails, all browsers using it are retired and new ones, each with its own new random IP, are started.
+
+```marked-tabs
+<marked-tab header="PuppeteerCrawler" lang="javascript">
+const Apify = require("apify");
+
+Apify.main(async () => {
+    const requestList = await Apify.openRequestList(
+        "my-list", ["http://www.example.com"]
+    );
+    const proxyConfiguration = await Apify.createProxyConfiguration({
+        // if you need to use more then one group
+        // simply add the additional ID to the array below
+        groups: ["SHADER"],
+    });
+
+    const crawler = new Apify.PuppeteerCrawler({
+        requestList,
+        proxyConfiguration,
+        useSessionPool: true,
+        handlePageFunction: async ({ page, request, proxyInfo }) => {
+            return Apify.pushData({ title: await page.title() });
+        },
+        handleFailedRequestFunction: ({ request }) => {
+            console.error("Request failed", request.url, request.errorMessages);
+        },
+    });
+
+    await crawler.run();
+});
+</marked-tab>
+``` 
+
+Using [PuppeteerCrawler](https://sdk.apify.com/docs/api/puppeteer-crawler#docsNav), keep a single IP address selected from the `SHADER` proxy group until it fails (gets retired).
+
+The `maxPoolSize: 1` configuration means that a single IP will be used by all browsers until it fails. Then, all running browsers are retired, a new IP is selected and new browsers opened. The browsers all use the new IP.
+
+```marked-tabs
+<marked-tab header="PuppeteerCrawler" lang="javascript">
+const Apify = require("apify");
+
+Apify.main(async () => {
+    const requestList = await Apify.openRequestList(
+        "my-list", ["http://www.example.com"]
+    );
+    const proxyConfiguration = await Apify.createProxyConfiguration({
+        // if you need to use more then one group
+        // simply add the additional ID to the array below
+        groups: ["SHADER"],
+    });
+
+    const crawler = new Apify.PuppeteerCrawler({
+        requestList,
+        proxyConfiguration,
+        useSessionPool: true,
+        sessionPoolOptions: {
+            sessionOptions: { maxPoolSize: 1 },
+        },
+        handlePageFunction: async ({ page, request, proxyInfo }) => {
+            return Apify.pushData({ title: await page.title() });
+        },
+        handleFailedRequestFunction: ({ request }) => {
+            console.error("Request failed", request.url, request.errorMessages);
+        },
+    });
+
+    await crawler.run();
+});
+</marked-tab>
+```
+
+### [](#use-a-single-ip-address-from-specific-groups-for-multiple-requests) Use a single IP address from specific groups for multiple requests
```

Comment:
I think even the title says that this is becoming too complex.

Let's separate the Group examples and Proxy rotation examples. The groups are rather simple, so perhaps only 2 examples are needed, groups and no groups (auto). The IP rotation examples can then just use auto and not get polluted with groups.